test/unit/custom_spec.js

```diff
@@ -0,0 +1,99 @@
+import { buildGetDocumentParams } from './test_utils';
+import { DOMCanvasFactory } from '../../src/display/dom_utils';
+import { getDocument } from '../../src/display/api';
+import { isNodeJS } from '../../src/shared/util';
+import { PDFJS } from '../../src/display/global';
+
+function getTopLeftPixel(canvasContext) {
+  var imgData = canvasContext.getImageData(0, 0, 1, 1);
+  var r = imgData.data[0];
+  var g = imgData.data[1];
+  var b = imgData.data[2];
+  var a = imgData.data[3];
+  return { r, g, b, a};
+}
+
+describe('custom canvas rendering', function() {
+  var transparentFileName = 'transparent.pdf';
+  var transparentGetDocumentParams = buildGetDocumentParams(transparentFileName);
```

Comment:
The `transparent.pdf` file isn't added to the patch, which cause failures in all the tests below. Also, since this line is too long, let's just do this instead:
```javascript
var transparentGetDocumentParams = buildGetDocumentParams('transparent.pdf');
```